lib/src/component2_suggestors/componentwillmount_migrator.dart

```diff
@@ -38,40 +40,92 @@ class ComponentWillMountMigrator extends GeneralizingAstVisitor
       return;
     }
 
-    if (extendsComponent2(containingClass)) {
-      // Update method name.
-      if (node.name.name == 'componentWillMount') {
+    MethodDeclaration componentDidMountMethodDecl =
+        containingClass.members.firstWhere(
+      (member) =>
+          member is MethodDeclaration &&
+          member.name.name == 'componentDidMount',
+      orElse: () => null,
+    );
+
+    if (extendsComponent2(containingClass) &&
+        node.name.name == 'componentWillMount') {
+      if (componentDidMountMethodDecl != null) {
+        if (node.body is BlockFunctionBody) {
+          bool hasSuperCall = componentDidMountMethodDecl.body
+              .toSource()
+              .contains('super.componentDidMount();');
+          String methodBody =
+              (node.body as BlockFunctionBody).block.statements.join('\n');
+
+          // Update or remove super call.
+          if (methodBody.contains('super.componentWillMount();')) {
+            methodBody = methodBody.replaceAll(
+              'super.componentWillMount();\n',
+              hasSuperCall ? '' : 'super.componentDidMount();\n',
+            );
+          }
+
+          // Move body of `componentWillMount` to end of `componentDidMount`.
+          yieldPatch(
+            componentDidMountMethodDecl.body.endToken.offset,
+            componentDidMountMethodDecl.body.endToken.offset,
+            methodBody,
+          );
+        }
+
+        // Copy any annotations not already present to `componentDidMount`.
+        String annotationsToAdd = '';
+        node.metadata.forEach((annotation) {
```

Comment:
#nit for-in loops should be preferred over forEach

https://dart.dev/guides/language/effective-dart/usage#avoid-using-iterableforeach-with-a-function-literal